soaktest/soakclient_transaction/soakclient_transaction.go

```
@@ -0,0 +1,495 @@
+				soaktest.ConsumerMsgErrs += 1
+				soaktest.DatadogIncrement(soaktest.CONSUMER_RECEIVE_ERROR_FROM+
+					inputTopic, 1)
+				fmt.Printf("kafka.Error: %v: %v for consumer %v\n",
+					e.Code(), e, c)
+
+			default:
+				fmt.Printf("Ignored %s for consumer %v\n", e, c)
+			}
+		}
+	}
+	return terminate(err, *c)
+}
+
+// Transaction producer, produce messages to output topic
+func produceMessage(e *kafka.Message,
+	p kafka.Producer,
+	msgid uint64,
+	outTopic string) {
+	encodingTime, err := time.Now().GobEncode()
+	if err != nil {
+		fmt.Errorf("Failed to encode current time %s\n", err)
+	}
+
+	err = p.Produce(&kafka.Message{
+		TopicPartition: kafka.TopicPartition{
+			Topic:     &outTopic,
+			Partition: kafka.PartitionAny},
+		Value: []byte(string(e.Value)),
+		Headers: []kafka.Header{{Key: "time",
+			Value: encodingTime},
+			{Key: "msgid",
+				Value: soaktest.ConvertUint64ToByteArray(msgid)}},
+	}, nil)
+	// Wait for delivery report
+	ev := <-p.Events()
+	switch e := ev.(type) {
+	case *kafka.Message:
+		soaktest.ProducerDeliveryCheck(e)
+
+	case *kafka.Stats:
+		soaktest.HandleStatsEvent(e, outTopic)
+
+	case kafka.Error:
+		// kafka.Errors should generally be
+		// considered informational, the client
+		// will try to automatically recover.
+		soaktest.DrErrCnt += 1
+		soaktest.DatadogIncrement(soaktest.PRODUCER_DR_ERR_TO+
+			outTopic, 1)
+		fmt.Printf("kafka.Error: %v: %v for producer %v\n",
+			e.Code(), e, p)
+
+	default:
+		fmt.Printf("Ignored event: %s for producer %v\n", ev, p)
+	}
+}
+
+// Commit transaction to output topic
+// Abort transaction once every 100 commit transactions
+func commitTransaction(partitionPosition []kafka.TopicPartition,
+	p kafka.Producer,
+	c kafka.Consumer,
+	transactionCommitNum int) (commitNum int, err error) {
+	fmt.Printf("=== Committing transaction===\n")
+
+	cgmd, err := c.GetConsumerGroupMetadata()
+	if err != nil {
+		fmt.Printf("Failed to get Consumer Group Metadata %v\n", err)
+	}
```

error is not handled